docs/guide/extensibility/indentation.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Indentation
+---
+
+# Indentation
+
+Sublime Text can perform automatic indentation when you type.
+It can also reindent text according to the indentation rules on demand.
+Note that it will not perform auto-formatting, like adding newlines or adjusting spaces around parenthesis and the like. You would need an external tool or plugin for that.
+The indentation rules consist of various regular expression patterns applied in different circumstances.
+All the regular expression patterns match against one line of text.
+
+## File Format
+
+Indentation rules are defined using metadata files.
+However, because metadata for indentation rules is complicated and
+is commonly required by packages which add syntax definitions,
+it's discussed separately in this page
+for convenience.
+
+Just as regular metadata files,
+indentation metadata files
+have the `.tmPreferences` extension
+and use the Property List format.
+The file name is ignored by Sublime Text.
+
+::: info See Also
+[Metadata](./metadata.md#indentation-options) Detailed documentation on indentation metadata.
+:::
+
+The scope selector is equivalent to the `eol_selector` context, wherein it works on the scope of the `\n` character at the end of the line.
+
+## Indentation when pressing <kbd>Enter</kbd>
+
+The main aspect of auto indentation increasing the indent level occurs when pressing <kbd>Enter</kbd>, anywhere on a line.
+Note that some keybindings override <kbd>Enter</kbd> with their own logic by inserting a snippet etc.
+The `insert` command processes indentation logic, but the `insert_snippet` and `append` commands don't.
+
+## All subsequent lines
+
+Example: type an open HTML tag `<article>` or curly brace `{` and press <kbd>Enter</kbd>
+This is handled by the `increaseIndentPattern` regex.
+
+## A single line
+
+Sometimes it is useful to indent just a single line further, for example when typing an if statement without an opening brace `if (x)` <kbd>Enter</kbd>. In this case, only the next line is indented, because the "if" will apply to only a single statement/line. Then, if a `{` is typed, it gets unindented again, because most people want their opening and closing braces in the same column as the start of the "if". See also the "Adjusting indentation when you type on a line" section.
+
+This is also potentially useful for incomplete statements (split over multiple lines), for example some languages require statements to be terminated with a semi-colon `;`, and visually indenting them makes it more readable.
+```js
+    a.b().c.d.e.f
+        .g().h.i();
```

Comment:
This is actually a bad example here because this exact use case breaks when a statement is broken over more than just two lines. A better idea would be to cover C-style bracket-less if statements.